plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -815,6 +819,21 @@ public Optional<ConnectorTableLayout> getNewTableLayout(ConnectorSession session
         return getWriteLayout(schema, partitionSpec, false);
     }
 
+    @Override
+    public Optional<io.trino.spi.type.Type> getSupportedType(ConnectorSession session, io.trino.spi.type.Type type)
+    {
+        if (type instanceof TimestampWithTimeZoneType) {
+            return Optional.of(createTimestampWithTimeZoneType(6));
```

Comment:
```suggestion
return Optional.of(TIMESTAMP_TZ_MICROS);
```

Reuse `io.trino.spi.type.TimestampWithTimeZoneType#TIMESTAMP_TZ_MICROS`